realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -361,6 +384,17 @@ private void emitValidateTableMethod(JavaWriter writer) throws IOException {
                         fieldTypeSimpleName, fieldName);
                 writer.endControlFlow();
 
+                // make sure that nullability matches
+                if (metadata.isNullable(field)) {
+                    writer.beginControlFlow("if (!table.isColumnNullable(%s))", staticFieldIndexVarName(field));
+                    writer.emitStatement("throw new RealmMigrationNeededException(transaction.getPath(), \"Add annotation @Required or @PrimaryKey to field '%s'\")", fieldName);
```

Comment:
2 things here:
1). Error message seems wrong. Table is not nullable even though it should be. This indicate primitive type used instead of boxed type? If `@Required` is used it wouldn't be nullable?

2) Error message should start with what is wrong then solution, e.g. "Field foo is not nullable. Add annotation @Required ...". Note error might need to be rewritten pr. 1)